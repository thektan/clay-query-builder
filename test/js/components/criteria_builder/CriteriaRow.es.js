import React from 'react';
import CriteriaRow from 'components/criteria_builder/CriteriaRow.es';
import {cleanup, render} from 'react-testing-library';

const connectDnd = jest.fn(el => el);

describe(
	'CriteriaRow',
	() => {
		afterEach(cleanup);

		it(
			'should render',
			() => {
				const OriginalCriteriaRow = CriteriaRow.DecoratedComponent;

				const {asFragment} = render(
					<OriginalCriteriaRow
						connectDragPreview={connectDnd}
						connectDragSource={connectDnd}
						connectDropTarget={connectDnd}
						groupId="group_01"
						index={0}
						onAdd={jest.fn()}
						onChange={jest.fn()}
						onDelete={jest.fn()}
						onMove={jest.fn()}
					/>
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);
	}
);