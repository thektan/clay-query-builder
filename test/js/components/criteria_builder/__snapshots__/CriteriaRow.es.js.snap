// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriteriaRow should render 1`] = `
<DocumentFragment>
  <div
    class="criterion-row-root"
  >
    <div
      class="edit-container"
    >
      <div
        class="drag-icon"
      >
        <svg
          aria-hidden="true"
          class="lexicon-icon lexicon-icon-drag"
          viewBox="0 0 512 512"
        >
          <use
            xlink:href="#drag"
          />
        </svg>
      </div>
      <span
        class="criterion-string"
      >
        <span />
         with property 
        <b />
      </span>
      <select
        class="form-control criterion-input operator-input"
      />
      <input
        class="criterion-input form-control"
        type="text"
        value=""
      />
      <button
        class="btn btn-outline-secondary btn-monospaced btn-outline-borderless duplicate-button"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lexicon-icon lexicon-icon-paste icon"
          viewBox="0 0 512 512"
        >
          <use
            xlink:href="#paste"
          />
        </svg>
      </button>
      <button
        class="btn btn-outline-secondary btn-monospaced btn-outline-borderless"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lexicon-icon lexicon-icon-trash icon"
          viewBox="0 0 512 512"
        >
          <use
            xlink:href="#trash"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
